sqlglot/dialects/bigquery.py

```diff
@@ -517,6 +513,16 @@ class Generator(generator.Generator):
             "within",
         }
 
+        def attimezone_sql(self, expression: exp.AtTimeZone) -> str:
+            if not isinstance(expression.parent, exp.Cast):
+                return self.func(
+                    "TIMESTAMP", self.func("DATETIME", expression.this, expression.args.get("zone"))
+                )
+            return super().attimezone_sql(expression)
```

Comment:
This was changed because we'd get the following inside of a cast expression:

```sql
TIMESTAMP(DATETIME('YYYY-MM-DD HH24:MI:SS TZH:TZM', 'Asia/Kolkata'))
```

This is invalid syntax in the context of a `cast(.. as string [format <fmt> [at time zone <tz>]])`.